Update to 5.x

volha-pivavarchyk opened the pull request:
<!-- ## Which branch should I use for my PR?

Assuming that:

a = current major release
b = current minor release
c = future major release

* a.x for any features and enhancements (e.g. 4.x)
* a.b for any bug fixes (e.g. 4.0, 4.1, 4.2)
* c.x for any features, enhancements or bug fixes with backward compatibility breaking changes (e.g. 5.x) -->

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] <!-- All PRs must maintain or improve code coverage -->
| Related user documentation PR URL      | mautic/mautic-documentation#... <!-- required for new features -->
| Related developer documentation PR URL | mautic/developer-documentation#... <!-- required for developer-facing changes -->
| Issue(s) addressed                     | Fixes #... <!-- prefix each issue number with "Fixes #", no need to create an issue if none exists, explain below instead -->

<!--
Additionally (see https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#work-on-your-pull-request):
 - Always add tests and ensure they pass.
 - Bug fixes must be submitted against the lowest maintained branch where they apply
   (lowest branches are regularly merged to upper ones so they get the fixes too.)
 - Features and deprecations must be submitted against the "4.x" branch.
-->

#### Description:

Replaces PRs #298, #308, #316
<!--
Please write a short README for your feature/bugfix. This will help people understand your PR and what it aims to do. If you are fixing a bug and if there is no linked issue already, please provide steps to reproduce the issue here.
-->

#### Steps to test this PR:

<!--
This part is really important. If you want your PR to be merged, take the time to write very clear, annotated and step by step test instructions. Do not assume any previous knowledge - testers may not be developers.
-->
1. Open this PR for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.

<!--
If you have any deprecations, list them here along with the new alternative.
If you have any backwards compatibility breaks, list them here.
-->

escopecz committed:
Adding GH Actions config

escopecz committed:
Adding PHPSTAN Baseline for existing issues

escopecz committed:
Remove dead code flagged by Rector

escopecz committed:
Fixing PHPUNIT config

escopecz committed:
Debugging tests

escopecz committed:
PHPSTAN baseline for each PHP version

escopecz committed:
Fixing tests

escopecz committed:
Some tests require theofidry/alice-data-fixtures

dongilbert committed:
Merge pull request #325 from acquia/MAUT-10760

Fixes for PHPStan & PHP CS Fixer

dongilbert committed:
PHPStan fix again

dongilbert committed:
PHPStan

dongilbert committed:
PHPStan

dongilbert committed:
PHPStan

tejasync committed:
Update README.md

tejasync committed:
Merge pull request #326 from acquia/update-readme-for-workflow

Update README.md

dongilbert committed:
Remove PUT from the available item operations for the custom_objects endpoint.

rohitp19 committed:
Merge pull request #327 from acquia/MAUT-10984

Remove PUT from the available item operations for the custom_objects endpoint

rahuld-dev committed:
as PUT method is removed we are using PATCH insted

dadarya0 committed:
Merge pull request #328 from acquia/MAUT-10984-tests

Fix test case

tejasync committed:
Merge remote-tracking branch 'origin/staging' into MAUT-8990

dadarya0 committed:
fix segement filter issue in custom object

dadarya0 committed:
fix test cases

dadarya0 committed:
add code coverage

dadarya0 committed:
fix phpstan issue

dadarya0 committed:
fix failing test case

rahuld-dev committed:
fixing patch request

rahuld-dev committed:
skip setDefaultValuesForMissingFields for patch request

rahuld-dev committed:
adding comment

dadarya0 committed:
Merge pull request #331 from acquia/MAUT-11028

MAUT-11028 : fix segement filter issue in custom object

rahuld-dev committed:
updating note for implementation

tejasync committed:
Merge remote-tracking branch 'origin/staging' into MAUT-8990

tejasync committed:
Fix warnings in test suite for PHP8.0 upgrade

tejasync committed:
Remove unused import

Co-authored-by: Saurabh Gupta <48244990+dadarya0@users.noreply.github.com>

rahuld-dev committed:
Merge pull request #332 from acquia/MAUT-10983

MAUT-10983 : fix for custom item PATCH request is working same as PUT call

tejasync committed:
Merge remote-tracking branch 'origin/staging' into MAUT-8990

tejasync committed:
Add merge filter query

tejasync committed:
Remove const is defined check

tejasync committed:
Fix test case by adding config param

himanshuprajapati-1 committed:
Merge pull request #319 from acquia/MAUT-8990

MAUT-8990 : Upgrade to PHP8

rahuld-dev committed:
check lead exist before link to custom item

rahuld-dev committed:
fixed tests

rahuld-dev committed:
adding tests

rahuld-dev committed:
removed php8 only compatibal code

rahuld-dev committed:
change warning type

rahuld-dev committed:
correct message

rahuld-dev committed:
changed warning type

rahuld-dev committed:
fix tests

rahuld-dev committed:
check for method exists and csfix

dadarya0 committed:
Merge pull request #335 from acquia/MAUT-11097

MAUT-11097 : check if lead exist before link to custom item or not

tejasync committed:
Add function to check if custom object filters are mergable

tejasync committed:
Revert CustomFieldFilterQueryBuilder

tejasync committed:
Update services

tejasync committed:
Add CustomObjectMergedFilterQueryBuilder

tejasync committed:
Use CustomObjectMergedFilterQueryBuilder

tejasync committed:
Check for custom item filter

tejasync committed:
Change merge filter logic

tejasync committed:
Make consts

tejasync committed:
Fix test cases

tejasync committed:
Test with custom_object_merge_filter param

tejasync committed:
Move test to appropriate class name

tejasync committed:
Rename test case

tejasync committed:
Rename test case

tejasync committed:
Revert unnecessary code

tejasync committed:
Remove union return type declaration in favour of PHP7.4

tejasync committed:
Remove not exists in favour of negative filters

tejasync committed:
Fix contact not getting added when one segment filter is used

volha-pivavarchyk committed:
Merge branch 'feature/adapt-to-mautic-v5-Twig-to-merge' into update-to-5x

volha-pivavarchyk committed:
Merge branch 'feature/adapt-to-mautic-v5-transform-controllers' into update-to-5x

# Conflicts:
#	Config/config.php
#	Controller/CustomField/SaveController.php
#	Controller/CustomObject/DeleteController.php
#	Tests/Unit/Controller/ControllerTestCase.php

volha-pivavarchyk committed:
fix dependencies

volha-pivavarchyk committed:
add parent constructor call

byt3sage commented:
@volha-pivavarchyk this looks amazing! Will test it this week and provide anything I spot here :)

byt3sage commented:
@volha-pivavarchyk clicking 'Custom Objects' on the left hand sidebar and clicking into a custom object doesn't do anything :)

Going to the URL manually gives this error:
```
[2024-02-22T12:18:27.316546+00:00] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\InvalidFieldNameException: "An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.is_unique_identifier' in 'field list'" at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php line 67 {"exception":"[object] (Doctrine\\DBAL\\Exception\\InvalidFieldNameException(code: 1054): An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.is_unique_identifier' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:67)\n[previous exception] [object] (Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1054): SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.is_unique_identifier' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28)\n[previous exception] [object] (PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.is_unique_identifier' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:130)"} {"hostname":"ip-172-31-40-170","pid":668126}
[2024-02-22T12:18:27.317428+00:00] mautic.CRITICAL: Exception thrown when handling an exception (InvalidArgumentException: The HTTP status code "1054" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php line 470) {"exception":"[object] (InvalidArgumentException(code: 0): The HTTP status code \"1054\" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php:470)"} {"hostname":"ip-172-31-40-170","pid":668126}
[2024-02-22T12:18:27.322066+00:00] mautic.ERROR: InvalidArgumentException: The HTTP status code "1054" is not valid. - in file /var/www/mautic/vendor/symfony/http-foundation/Response.php - at line 470 [] {"hostname":"ip-172-31-40-170","pid":668126}
```

volha-pivavarchyk commented:
@jaetoole Thank you. I'll fix. I'll test other features too. But I can do it  over the weekend. I let you know when I'm done.

byt3sage commented:
@volha-pivavarchyk thanks! It looks like the plugin schema isn't getting updated, I can see the migrations to update that field but it doesn't run unless I install the plugin fresh without the custom_ tables been in there

escopecz commented:
That should be solved by bumping the version in https://github.com/acquia/mc-cs-plugin-custom-objects/blob/staging/Config/config.php#L13. The migrations run only if the version is higher.

byt3sage commented:
Ah ace - @volha-pivavarchyk did you manage to sort this or would you like me to fix?

volha-pivavarchyk commented:
@jaetoole 

Sorry, I haven't had any time at all. I had to work on an urgent project task. I'll be able to fix it as soon as I'm done. I have a limited amount of time right now as I'm working on another project. 

I don't mind if you fix it. But I think there will be more issues that will come up. It may be easier for me to fix this one and others since I spent a lot of time earlier. I'll try to do it tonight I hope to finish the current task. 

However, you can pick up PR or we can work together. It's up to you. Anything works for me.

volha-pivavarchyk committed:
remove duplicate dependencies; change deprecated methods

byt3sage commented:
@volha-pivavarchyk I'll let you take the lead on this as I'm fairly new to Mautic. If there is anything you think is fairly easy and want to delegate to me then just give me a shout :)

volha-pivavarchyk committed:
fix inserting variable into string

volha-pivavarchyk committed:
fix an event dispatching issue

volha-pivavarchyk committed:
fix import with contact ids

volha-pivavarchyk committed:
fix the allowed memory size exhausted issue

volha-pivavarchyk committed:
fix the inheritance model issue

tejasync committed:
Merge pull request #333 from acquia/MAUT-11028-2

Fix segment build when multiple custom items are linked to a contact

volha-pivavarchyk commented:
@jaetoole Thank you. I would be more comfortable being a member of the led by @escopecz. I just want to contribute what we've done. Let's work together on it. 🙂

To finish this PR I think we need 
 1. Test carefully. I didn't test export/import and campaign yet.
 2. Adapt to new form processing. (I've done it for us and just need to add here)
 3. Investigate if we can improve or get rid of the config file.
 4. Fix tests. I started to do it but didn't add here yet. It might be another PR.
 
5. Then we can simplify and a bit improve the code using constructor property promotion, type hinting everywere, phpdoc blocks, etc. I think it should be another PR.

If you test the project it would be perfect. I'll add the code to adapt to new form processing. Whoever makes it first takes point 3.
WDYT?

@escopecz WDYT about points 4 and 5 and all of that?

escopecz commented:
@volha-pivavarchyk firstly, thank you very much for this work. It's much appreciated even though it's not visible from our side. I agree with all the points except the point 4. The tests should be passing in order to get this merged. That's how you know the PR did not break anything.

volha-pivavarchyk commented:
@jaetoole 
> @volha-pivavarchyk clicking 'Custom Objects' on the left hand sidebar and clicking into a custom object doesn't do anything :)
> 
> Going to the URL manually gives this error:
> 
> ```
> [2024-02-22T12:18:27.316546+00:00] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\InvalidFieldNameException: "An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.is_unique_identifier' in 'field list'" at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php line 67 {"exception":"[object] (Doctrine\\DBAL\\Exception\\InvalidFieldNameException(code: 1054): An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.is_unique_identifier' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:67)\n[previous exception] [object] (Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1054): SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.is_unique_identifier' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28)\n[previous exception] [object] (PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.is_unique_identifier' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:130)"} {"hostname":"ip-172-31-40-170","pid":668126}
> [2024-02-22T12:18:27.317428+00:00] mautic.CRITICAL: Exception thrown when handling an exception (InvalidArgumentException: The HTTP status code "1054" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php line 470) {"exception":"[object] (InvalidArgumentException(code: 0): The HTTP status code \"1054\" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php:470)"} {"hostname":"ip-172-31-40-170","pid":668126}
> [2024-02-22T12:18:27.322066+00:00] mautic.ERROR: InvalidArgumentException: The HTTP status code "1054" is not valid. - in file /var/www/mautic/vendor/symfony/http-foundation/Response.php - at line 470 [] {"hostname":"ip-172-31-40-170","pid":668126}
> ```

I think the migrations have not been run. Check it in DB. As I figured out earlier migrations are run during plugin installation. 
mautic:plugins:install,
mautic:plugins:update,
or
mautic:plugins:reload = plugins:install + mautic:plugins:update
or 
right panel -> plugins -> install/upgrade plugins

> That should be solved by bumping the version in https://github.com/acquia/mc-cs-plugin-custom-objects/blob/staging/Config/config.php#L13. The migrations run only if the version is higher.

Maybe this way is better I've never try it

volha-pivavarchyk commented:
> @volha-pivavarchyk firstly, thank you very much for this work. It's much appreciated even though it's not visible from our side. I agree with all the points except the point 4. The tests should be passing in order to get this merged. That's how you know the PR did not break anything.

@escopecz  You are right. I or we will fix the tests

byt3sage commented:
> @jaetoole
> 
> > @volha-pivavarchyk clicking 'Custom Objects' on the left hand sidebar and clicking into a custom object doesn't do anything :)
> > Going to the URL manually gives this error:
> > ```
> > [2024-02-22T12:18:27.316546+00:00] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\InvalidFieldNameException: "An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.is_unique_identifier' in 'field list'" at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php line 67 {"exception":"[object] (Doctrine\\DBAL\\Exception\\InvalidFieldNameException(code: 1054): An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.is_unique_identifier' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:67)\n[previous exception] [object] (Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1054): SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.is_unique_identifier' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28)\n[previous exception] [object] (PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.is_unique_identifier' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:130)"} {"hostname":"ip-172-31-40-170","pid":668126}
> > [2024-02-22T12:18:27.317428+00:00] mautic.CRITICAL: Exception thrown when handling an exception (InvalidArgumentException: The HTTP status code "1054" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php line 470) {"exception":"[object] (InvalidArgumentException(code: 0): The HTTP status code \"1054\" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php:470)"} {"hostname":"ip-172-31-40-170","pid":668126}
> > [2024-02-22T12:18:27.322066+00:00] mautic.ERROR: InvalidArgumentException: The HTTP status code "1054" is not valid. - in file /var/www/mautic/vendor/symfony/http-foundation/Response.php - at line 470 [] {"hostname":"ip-172-31-40-170","pid":668126}
> > ```
> 
> I think the migrations have not been run. Check it in DB. As I figured out earlier migrations are run during plugin installation. mautic:plugins:install, mautic:plugins:update, or mautic:plugins:reload = plugins:install + mautic:plugins:update or right panel -> plugins -> install/upgrade plugins
> 
> > That should be solved by bumping the version in https://github.com/acquia/mc-cs-plugin-custom-objects/blob/staging/Config/config.php#L13. The migrations run only if the version is higher.
> 
> Maybe this way is better I've never try it

Yeah, I think we need to bump the plugin version so it triggers an update to run the migrations

escopecz committed:
Handling custom item links when merging contacts

escopecz committed:
Merge remote-tracking branch 'origin' into MAUT-11209

aadarshjain-dev committed:
Test gh-actions

aadarshjain-dev committed:
tests.yaml update

aadarshjain-dev committed:
PHPSTAN baseline (debug) update

escopecz committed:
Merge pull request #340 from acquia/MAUT-11209

Handling custom item links when merging contacts

aadarshjain-dev committed:
Comment PHPUNIT TESTS

aadarshjain-dev committed:
Comment Coverage Report step

aadarshjain-dev committed:
skip test

aadarshjain-dev committed:
skip test

aadarshjain-dev committed:
revert

aadarshjain-dev committed:
print commit id

aadarshjain-dev committed:
Push commit to mc-cs

aadarshjain-dev committed:
Push commit to mc-cs

aadarshjain-dev committed:
Push commit to mc-cs

aadarshjain-dev committed:
Push commit to mc-cs

aadarshjain-dev committed:
Push commit to mc-cs

aadarshjain-dev committed:
Push commit to mc-cs

GuzmanBellon commented on Entity/CustomFieldOption.php:
```diff
@@ -164,31 +163,31 @@ public function setOrder(?int $order): void
      * {@inheritdoc}
      */
-    public function offsetGet($offset)
+    public function offsetGet(mixed $offset): bool
     {
         return $this->offsetExists($offset) ? $this->{$offset} : null;
     }
```
The return type declaration breaks the function. e.g. `$this->label` returns a `string`
```suggestion
    public function offsetGet(mixed $offset): mixed
    {
        return $this->offsetExists($offset) ? $this->{$offset} : null;
    }
```

GuzmanBellon reviewed (commented):
Hi Volha.

Thanks for your efforts to port the plugin to 5.x. 
I'm brand new to Mautic but by testing your PR I found a problem wenn adding a select field to a new custom object. When you create a new option you get an error.
```
MauticPlugin\CustomObjectsBundle\Entity\CustomFieldOption::offsetGet(): Return value must be of type bool, string returned
```
![image](https://github.com/acquia/mc-cs-plugin-custom-objects/assets/47144569/5ded02a3-3367-45dc-ba4a-76c617c23d69)

aadarshjain-dev committed:
static analysis add

aadarshjain-dev committed:
Moving Install composer up and comment phpstan baseline

aadarshjain-dev committed:
Upgrade php and mysql version

aadarshjain-dev committed:
Uncoomment PHPSTAN step

aadarshjain-dev committed:
comment phpstan

escopecz committed:
Debug: regenerating PHPSTAN baseline

escopecz committed:
Updating PHPSTAN baseline, enabling PHPSTAN check

escopecz committed:
Clearing up the baseline files to generate them from scratch

escopecz committed:
Running PHPSTAN with fresh baselines

escopecz committed:
Updating action versions to avoid NodeJS warnings

escopecz committed:
Removing the pull_request settings hoping the CI will run on the commit without merging to staging

escopecz committed:
Disabling composer cache as theofidry/alice-data-fixtures doesn't seem to be present

escopecz committed:
Would it help to run the commands separately?

escopecz committed:
Or lock the version?

escopecz committed:
The dependency must be also set in kernel. Let's skip the tests that need it for now

escopecz committed:
The coverage report contains whole Mautic. Testing something different

escopecz committed:
Fixing kernel class

escopecz committed:
Another attempt to resolve the kernel class issue

escopecz committed:
Do we need to set KERNEL_CLASS?

fedys committed:
Check API request by the controller name

Commit:
fix customobject fetch from typeoparatorprovider

Commit:
Merge branch 'staging' into MAUT-11442

Commit:
split into 2 line

Commit:
var name change

fedys committed:
empty and neq operator fixed

fedys committed:
pass true for $filterAlreadyNegated only from CustomFieldFilterQueryBuilder

fedys committed:
CustomItemRelationQueryBuilderTestCase was not included within the test suite as its name didn't end with "Test"

fedys committed:
Added functional tests for "not equal" and "empty" operators

rahuld-dev committed:
adding KERNEL_CLASS

rahuld-dev committed:
fixing test

aarohiprasad committed:
fixed UI for between op, test case WIP

rahuld-dev committed:
fixing tests

aarohiprasad committed:
Functioning code and test case for custom item between query for numbers

aarohiprasad committed:
better handling of types

fedys committed:
Update Tests/Functional/Segment/Query/Filter/NegativeOperatorFilterQueryBuilderTest.php

Co-authored-by: Avikarsha Saha <avikarsha.saha@gmail.com>

fedys committed:
Merge pull request #343 from acquia/MAUT-11469

Check API request by the controller name instead of real request path

fedys committed:
Merge branch 'staging' into MAUT-11107-2

Commit:
Merge branch 'staging' into MAUT-11442

rahuld-dev committed:
enabling phpstan and fixed issues

aarohiprasad committed:
php7.4 syntax

aarohiprasad committed:
removed php 8 only syntax

rahuld-dev committed:
I think composer install should run after install mautic

aarohiprasad committed:
added test case for where assertion

rahuld-dev committed:
removed dependacy install after mautic installation

rahuld-dev committed:
removed vendor from scanDirectories from neon file

dadarya0 committed:
added between/not between operator for date type field

dadarya0 committed:
fix sonar issue

dadarya0 committed:
add functional test case

dadarya0 committed:
fix syntax

dadarya0 committed:
fix issue for development branch failing

rahuld-dev committed:
Adding inTheLast and inTheNext operator for dat and datetime type fields

fedys committed:
Fixed CampaignConditionTest

fedys committed:
phpstan-ignore

fedys committed:
phpstan-ignore

fedys committed:
Comments for workarounds

fedys committed:
Merge pull request #345 from acquia/MAUT-11107-2

NOT EQUAL operator not working with Custom Object Field filters

fedys committed:
Merge pull request #307 from acquia/gh-actions

Adding GH Actions config

fedys committed:
CS fixes

fedys committed:
NegativeOperatorFilterQueryBuilderTest fixed

fedys committed:
NegativeOperatorFilterQueryBuilderTest fixed

fedys committed:
Merge branch 'staging' into MAUT-11443

fedys committed:
Merge branch 'staging' into MAUT-11289

# Conflicts:
#	Helper/QueryFilterHelper.php

fedys committed:
Merge branch 'staging' into MAUT-11442

dadarya0 committed:
Update Helper/QueryFilterHelper.php

Co-authored-by: Aarohi Prasad <aarohi.prasad@acquia.com>

dadarya0 committed:
fixc s

dadarya0 committed:
fix date issue if empty/not empty filter is used

dadarya0 committed:
Merge remote-tracking branch 'origin/staging' into MAUT-11467

dadarya0 committed:
fix phpstan

dadarya0 committed:
fix failing test case

aarohiprasad committed:
removed extra parameter

aarohiprasad committed:
syntax changes

fedys committed:
Env variable MAUTIC_PROJECT_VERSION introduced

fedys committed:
TypeError : str_starts_with() fixed

dadarya0 committed:
fix same for merged filter too

dadarya0 committed:
change variable names

aarohiprasad committed:
fixed failing test

dadarya0 committed:
Merge pull request #349 from acquia/MAUT-11467

MAUT-11467 : fix date issue if empty/not empty filter is used in segment filter

dadarya0 committed:
Merge remote-tracking branch 'origin/staging' into MAUT-11289

dadarya0 committed:
fix not beetween issue

escopecz committed:
Merge pull request #347 from acquia/MAUT-11289

MAUT-11289 : Segments: New (not)between date filter

Commit:
Merge branch 'staging' into MAUT-11287

avikarshasaha committed:
Merge pull request #346 from acquia/MAUT-11287

Between operator for segments with numeric Custom Object fields

Commit:
Merge branch 'staging' into MAUT-11442

Commit:
skip for 4.4

Commit:
skip for 4.4 csfix

Commit:
skip for 4.4 csfix

Commit:
skip for 4.4 csfix

Commit:
7.4 syntax

Commit:
stan fix

Commit:
stan fix

Commit:
stan fix

Commit:
fixes phpstan

Commit:
skip phpstan

Commit:
removing ingnore comment

avikarshasaha committed:
Merge pull request #344 from acquia/MAUT-11442

MAUT-11442: Before and After Filter in Segments | Fetch from TypeOperatorProvider

Commit:
removing unnecessary array filter

rahuld-dev committed:
Merge branch 'staging' into MAUT-11443

avikarshasaha committed:
Merge pull request #351 from acquia/MAUT-11442

Removing unnecessary array filter

rahuld-dev committed:
Merge pull request #348 from acquia/MAUT-11443

"in the last" and "in the next" segment filters.

escopecz commented:
Just a note that the default (staging) branch has now working GitHub Actions setup to test this plugin on Mautic 4. This PR should take the change and test it against the latest Mautic 5. This way the tests should uncover hidden issues with this update for Mautic 5.

escopecz committed:
Hide the new operators from campaign condition, leave them for segment fitlers

escopecz committed:
Fixing operators also for int field types

Commit:
cast string

Commit:
test

avikarshasaha committed:
Merge pull request #354 from acquia/MAUT-8476

MAUT-8476: Integer in label for option error fix

rahuld-dev committed:
Merge pull request #353 from acquia/MAUT-11602

Hide the new operators from campaign condition, leave them for segment filters

rahuld-dev committed:
MAUT-11616 : Fix operators do not refresh after selecting a different type of field in campaign event (#356)

* fix Operators do not refresh after selecting a different type of field

* csfix

* adding test which verify the data-operators attr is available

* fix phpstan

* fix tests

* suggessions

rahuld-dev committed:
MAUT-11617 : Campaign -> Condition -> Custom Object Field values : Date picker unavailable for Date type fields (#357)

* fix Operators do not refresh after selecting a different type of field

* csfix

* adding test which verify the data-operators attr is available

* fix phpstan

* fix tests

* datepicker for date and datetime fields

* using const type for fieldType

Co-authored-by: Miroslav Fedeleš <miroslav.fedeles@gmail.com>

* suggessions

* ifx value should not be empty after change operator

---------

Co-authored-by: Miroslav Fedeleš <miroslav.fedeles@gmail.com>

rohitp19 committed:
Replaced logger with monolog.logger.mautic (#355)

dadarya0 committed:
MAUT-11515 : Custom object date segment filters should have have same operators than date custom field filters (#358)

* added all date/datetime filters to custom oject date fields

* check if getcontext is exist

* Added new function instead changing old one

* change if condition sequence

* increase code coverage

* skip test case if mrunning on mautic 4.4

mehransobhani commented:
"I am having trouble installing this plugin on version 5.1. When will this issue be resolved?"

avikarshasaha committed:
Maut 11460: Dynamic content in email builder filtered on custom object values (#352)

* on replacement

* fix test

* test

* test

* test

* fix

* skiped for 4.4

* WIP

* refining

* test case

* test refine

* fix test case

* fix test case

* split lines

* CR

* unnecessary query remove

* check isset

* local cache

* key with id

* test cases

* Update EventListener/TokenSubscriber.php

Co-authored-by: John Linhart <admin@escope.cz>

* Update EventListener/TokenSubscriber.php

Co-authored-by: John Linhart <admin@escope.cz>

* WIP

* string concat

* match logic for CO

* test case

* test case csfix

* test case csfix

* const

* var in config

* fix null case for datetime

* cs fix

* fix for select and multi select

* for noraml lead field

* test case

* split line

* test case

* fix for like and select

---------

Co-authored-by: Avikarsha Saha <avikarsha.saha@acquia.com>
Co-authored-by: John Linhart <admin@escope.cz>

volha-pivavarchyk commented:
@mehransobhani Hi. Unfortunately, I'm very busy with my current project, so I don't have much free time. I'll try very hard to come back next week and fix the issues with this PR, adopting to v5.1.

letslingeronyou commented:
Thank you very much for the plugin. I'm really, really anxious to try this in Mautic 5.1.
I am considering downgrade Mautic just because of this plugin to work but i'll wait for your update... no pressure :)
thank you

fattahzadeh commented:
Hello
Please prepare this important plugin for version 5.1 as soon as possible, 
we will check here every day to see when it is ready to install.
Thank you

volha-pivavarchyk committed:
Update Entity/CustomFieldOption.php

Co-authored-by: GuzmanBellon <47144569+GuzmanBellon@users.noreply.github.com>

volha-pivavarchyk committed:
Fix EventDispatcher calling

volha-pivavarchyk committed:
Fix deprecated fetchColumn call

volha-pivavarchyk committed:
Use appropriate service to run a command in the test

volha-pivavarchyk committed:
Add #[AllowDynamicProperties] attribute to allow using dynamic properies

volha-pivavarchyk commented:
@GuzmanBellon Thanks for commit! I merged it

volha-pivavarchyk commented:
@nimah2o @letslingeronyou I apologize for the delay :). I have tested this branch with version 5.1. It works well for me. But this PR can't be merged until all the tests are fixed. I've been changing the code, but hardly fixing the tests. Most of it is due to class declaration and deprecated stuff. I fixed some, but it will take some time. I'll try to fix it in the next few days. Or will write that I'm running out of time :).

volha-pivavarchyk committed:
Fix tests of controllers

adamchemiawppl commented:
> @nimah2o @letslingeronyou I apologize for the delay :). I have tested this branch with version 5.1. It works well for me. But this PR can't be merged until all the tests are fixed. I've been changing the code, but hardly fixing the tests. Most of it is due to class declaration and deprecated stuff. I fixed some, but it will take some time. I'll try to fix it in the next few days. Or will write that I'm running out of time :).

Sorry for not adding much to discussion, but just wanted to let you know I'm waiting and I sincerely appreciate your work!

volha-pivavarchyk committed:
Fix test of controllers

volha-pivavarchyk committed:
Fix controller tests

volha-pivavarchyk committed:
Fix controller tests

volha-pivavarchyk committed:
Fix model tests

volha-pivavarchyk committed:
Fix listener tests

volha-pivavarchyk committed:
Fix Form tests

volha-pivavarchyk committed:
Fix controller tests

volha-pivavarchyk committed:
Fix repository tests

connorferster commented:
Amazing! Looking forward to the merge!

volha-pivavarchyk committed:
Fix functional tests

volha-pivavarchyk committed:
Fix functional tests

volha-pivavarchyk committed:
Add a required field to fixtures

volha-pivavarchyk committed:
Fix created objects for testing

volha-pivavarchyk commented:
Tests work well now.
![image](https://github.com/user-attachments/assets/3ad95583-011c-4ad2-934c-56e5f224277d)

I have yet to test manually to finish PR.

volha-pivavarchyk committed:
Fix pagination

volha-pivavarchyk committed:
Remove commented code

volha-pivavarchyk commented:
I manually tested the basic functions with the 5.x branch. Works well for me. Could someone also test it please.

RCheesley commented:
@connorferster @byt3sage @GuzmanBellon @mehransobhani @letslingeronyou @nimah2o @adamchemiawppl now's your opportunity to step up and test so that this can be merged and released!

adamchemiawppl commented:
Hey, installed it locally (local docker environment mostly same as our production setup), and everything seems to run fine:

- I can create new Custom Object via UI (will test API once I have more time). 
- Can create segment based on CO field filter. 
- Users get added based on that filter. 
- Can create and delete new Custom Object type with email field
- Custom object appears on relevant user page

I'm on quite busy schedule now, will test it further once I have some more time. If anything in particular should be tested - let me know.

@volha-pivavarchyk - thanks for your work and congrats on getting it done :-)

RCheesley commented:
@adamchemiawppl awesome thanks!

mehransobhani commented:
@volha-pivavarchyk 
I am unfortunately still having trouble installing this plugin.
When I add the plugin to Mautic, the Mautic goes down.

mehransobhani commented:
root@srv7393420426:/var/www/html/mautic# php bin/console cache clear
PHP Fatal error:  Uncaught Error: Class "MauticPlugin\CustomObjectsBundle\Provider\CustomFieldRouteProvider" not found in /var/www/html/mautic/plugins/mc-cs-plugin-custom-objects/Config/config.php:19
Stack trace:
#0 /var/www/html/mautic/app/bundles/CoreBundle/Loader/ParameterLoader.php(143): include()
#1 /var/www/html/mautic/app/bundles/CoreBundle/Loader/ParameterLoader.php(33): Mautic\CoreBundle\Loader\ParameterLoader->loadDefaultParameters()
#2 /var/www/html/mautic/app/AppKernel.php(358): Mautic\CoreBundle\Loader\ParameterLoader->__construct()
#3 /var/www/html/mautic/app/AppKernel.php(228): AppKernel->getParameterLoader()
#4 /var/www/html/mautic/vendor/symfony/framework-bundle/Console/Application.php(168): AppKernel->boot()
#5 /var/www/html/mautic/vendor/symfony/framework-bundle/Console/Application.php(74): Symfony\Bundle\FrameworkBundle\Console\Application->registerCommands()
#6 /var/www/html/mautic/vendor/symfony/console/Application.php(171): Symfony\Bundle\FrameworkBundle\Console\Application->doRun()
#7 /var/www/html/mautic/bin/console(16): Symfony\Component\Console\Application->run()
#8 {main}
  thrown in /var/www/html/mautic/plugins/mc-cs-plugin-custom-objects/Config/config.php on line 19
  ######################################################################
This error occurs for me
mautic : v5.1.0
php : v8.1.27
@volha-pivavarchyk

volha-pivavarchyk commented:
@mehransobhani It works well for me. 
![image](https://github.com/user-attachments/assets/54665a01-b2f8-4ee4-80ab-64da207f7582)

I can assume that the folder name for the package is incorrect. Could you try to clone the plugin to the folder `CustomObjectsBundle`.
![image](https://github.com/user-attachments/assets/4b24e460-f48e-4ef5-acfe-52ec7dd1324f)

Has it worked for you like this before?

Hugo-Prossaird commented:
@volha-pivavarchyk
I have identified a visual issue.

When you link a product to a contact and then navigate to the contact timeline, the event name is not displayed correctly:

![Capture d’écran 2024-08-29 à 09 44 42](https://github.com/user-attachments/assets/3417d588-da09-4183-886c-8b1d4ff2636c)

escopecz commented:
@volha-pivavarchyk were you able to merge in the changes from the staging branch? It should have the Github Action configuration in it. I don't see the CI jobs running on this PR.

mehransobhani commented:
oot@srv7393420426:/var/www/html/mautic#  php bin/console cache:clear

 // Clearing the cache for the prod environment with debug                      
 // false                                                                       

PHP Fatal error:  Access level to MauticPlugin\CustomObjectsBundle\Controller\CustomObject\FormController::renderForm() must be protected (as in class Symfony\Bundle\FrameworkBundle\Controller\AbstractController) or weaker in /var/www/html/mautic/plugins/CustomObjectsBundle/Controller/CustomObject/FormController.php on line 129

In FormController.php line 129:
                                                                               
  Compile Error: Access level to MauticPlugin\CustomObjectsBundle\Controller\  
  CustomObject\FormController::renderForm() must be protected (as in class Sy  
  mfony\Bundle\FrameworkBundle\Controller\AbstractController) or weaker        
                                                                               
                                                                              @volha-pivavarchyk

pars1455 commented:
Have you found a solution for the error I encountered?

> oot@srv7393420426:/var/www/html/mautic# php bin/console cache:clear
> 
> // Clearing the cache for the prod environment with debug // false
> 
> PHP Fatal error: Access level to MauticPlugin\CustomObjectsBundle\Controller\CustomObject\FormController::renderForm() must be protected (as in class Symfony\Bundle\FrameworkBundle\Controller\AbstractController) or weaker in /var/www/html/mautic/plugins/CustomObjectsBundle/Controller/CustomObject/FormController.php on line 129
> 
> In FormController.php line 129:
> 
> Compile Error: Access level to MauticPlugin\CustomObjectsBundle\Controller\ CustomObject\FormController::renderForm() must be protected (as in class Sy mfony\Bundle\FrameworkBundle\Controller\AbstractController) or weaker
> 
> ```
>                                                                           @volha-pivavarchyk 
> ```

hey , Have you found a solution for the error I encountered? I am waiting for your response
 @volha-pivavarchyk

mehransobhani commented:
> oot@srv7393420426:/var/www/html/mautic# php bin/console cache:clear
> 
> // Clearing the cache for the prod environment with debug // false
> 
> PHP Fatal error: Access level to MauticPlugin\CustomObjectsBundle\Controller\CustomObject\FormController::renderForm() must be protected (as in class Symfony\Bundle\FrameworkBundle\Controller\AbstractController) or weaker in /var/www/html/mautic/plugins/CustomObjectsBundle/Controller/CustomObject/FormController.php on line 129
> 
> In FormController.php line 129:
> 
> Compile Error: Access level to MauticPlugin\CustomObjectsBundle\Controller\ CustomObject\FormController::renderForm() must be protected (as in class Sy mfony\Bundle\FrameworkBundle\Controller\AbstractController) or weaker
> 
> ```
>                                                                           @volha-pivavarchyk 
> ```

hey , Have you found a solution for the error I encountered? I am waiting for your response
@volha-pivavarchyk

volha-pivavarchyk commented:
@Hugo-Prossaird I have checked this problem. It can only be fixed in Mautic. I can simply remove the bold object name, in which case it will look like this.
![image](https://github.com/user-attachments/assets/e6b2bc30-5109-468c-babd-9f9be229bb8a)

instead of:
![image](https://github.com/user-attachments/assets/03f23c90-a61c-40c3-8633-d2859d5d53f4)

@escopecz Which way is better. Change in Mautic the timeline event name display (allow some tags to be displayed) or remove `strong` tags in CustomObject.

volha-pivavarchyk commented:
> @volha-pivavarchyk were you able to merge in the changes from the staging branch? It should have the Github Action configuration in it. I don't see the CI jobs running on this PR.

Of course I can. I checked, there are a few conflicts. So I should be able to do it in the next few days. I apologize, sometimes I don't have time, so I don't respond immediately. If someone can do it faster, it's ok.

volha-pivavarchyk commented:
@mehransobhani Could you please check which branch do you use. The `renderForm` name of the method was used previously. Currently it's named `renderFormForObject`. 
https://github.com/acquia/mc-cs-plugin-custom-objects/blob/8de652b8c5ac9d25a38ea4a4f05b02722843cbf9/Controller/CustomObject/FormController.php#L117

And could you please check if there is a method `renderForm` in `MauticPlugin\CustomObjectsBundle\Controller\ CustomObject\FormController`. If so, could you please send me the code for this method so I can figure out what version of the class you have

mehransobhani commented:
@volha-pivavarchyk 
i now cloned the repository from https://github.com/acquia/mc-cs-plugin-custom-objects/tree/8de652b8c5ac9d25a38ea4a4f05b02722843cbf9
This is my first time installing this plugin, and after clearing the cache, I don't receive any errors, but Mautic becomes disabled as shown in the image below.
![mautic](https://github.com/user-attachments/assets/f6011dc5-8fcb-4aa6-8dcb-f2db05d3a293)

escopecz commented:
@volha-pivavarchyk 

> @escopecz Which way is better. Change in Mautic the timeline event name display (allow some tags to be displayed) or remove strong tags in CustomObject.

It should be fixed in Mautic using the`|purify` filter.

madalingavanarescu commented:
I'm trying to get the acquia/mc-cs-plugin-custom-objects plugin working on Mautic v5.1.0 but keep encountering errors. Any suggestions or an estimated timeline for a fix?

RCheesley commented:
When this PR is tested and merged, it'll be compatible with the latest release, if you can help with testing / fixing issues reported, that will help move it along.

dadarya0 committed:
MAUT-11383 : incorrect date value being calculating when the segment filter value is "yesterday" and operator is "gt" for date field (#359)

* Fix custom parameter date issue

* add more test case

volha-pivavarchyk commented:
> @volha-pivavarchyk were you able to merge in the changes from the staging branch? It should have the Github Action configuration in it. I don't see the CI jobs running on this PR.

@escopecz I tried merging changes from the staging branch, resolved conflicts, then some tests were broken. To be honest, I did this week ago and haven't had time to fix all the broken tests. Just a few of them. Because it takes more than a few hours. I guess I won't be able to get back to it until two weeks from now.

But I didn't see any changes in the Github Action configuration after the merge. So maybe the settings are the reason why the CI job for the test is not running. For example, branch naming, PR from forks. Can anyone check these settings as I am unable to do so. If yes, we can run this job for these changes and add changes from staging later.

![image](https://github.com/user-attachments/assets/c3d32fe5-ba84-4856-854d-17655fb9dae5)

volha-pivavarchyk commented:
@escopecz I have fixed 4 more tests. But there are still failed tests after the merge. Fixing them will take time, as it's due to big updates in the staging branch, which are not in this branch. I won't be able to get back to it until a week and a half from now. Do I need to push the merge and commits with the fixed tests after merging? Or can this PR be without the merge in this branch?

escopecz commented:
@volha-pivavarchyk it doesn't seem that the merge was complete. The [.github/workflows](https://github.com/acquia/mc-cs-plugin-custom-objects/tree/staging/.github/workflows) folder is missing from [this branch](https://github.com/volha-pivavarchyk/mc-cs-plugin-custom-objects/tree/update-to-5x/.github).

I don't fully understand what you are saying. You shared a screenshot that the merge commit is in the history. I thought you pushed it but now I checked it's not pushed. So the GH Actions won't run. Why do you think you should not push it?

volha-pivavarchyk committed:
Merge mautic-custom/staging into update-to-5x

volha-pivavarchyk committed:
Fix config after merging

volha-pivavarchyk committed:
Fix a variable name

volha-pivavarchyk committed:
Fix tests after merging

volha-pivavarchyk commented:
@escopecz I merged staging and fixed all tests. Works well for me. What next?

I apologize for my oversight. I thought that the workflows directory was  in the project.

escopecz commented:
@volha-pivavarchyk great news! I triggered the build and there seem to be some issue. Can you take a look? I will test this once the CI is passing.

volha-pivavarchyk committed:
Update dependencies

volha-pivavarchyk commented:
@escopecz I updated dependencies and workflow file (removed old php version and use new mautic version (v5.1)). Could you please trigger CI again.

escopecz commented:
Thanks! Triggered.

iuriJorbenadze commented:
Simplified Guide: Installing Custom Objects Plugin for Mautic 5.1.1
This guide helps you set up the Custom Objects plugin with Mautic 5.1.1 using the latest compatibility updates, while simplifying the upload process.
________________________________________

Al following steps are done locally in your computer before you even go to mautic instance.

Step 1: Clone the Plugin and Apply Compatibility Changes
1.	Clone the staging branch of the Custom Objects plugin using Git Bash:

`git clone -b staging https://github.com/acquia/mc-cs-plugin-custom-objects.git`

2.	Go into the cloned folder:

`cd mc-cs-plugin-custom-objects`

3.	Add the upstream repository:

`git remote add upstream https://github.com/acquia/mc-cs-plugin-custom-objects.git`

4.	Fetch the changes from PR #338 (Mautic 5 compatibility):

`git fetch upstream pull/338/head:pr-338`

5.	Merge the PR into your local staging branch:

```
git checkout staging
git merge pr-338
```

________________________________________
Step 2: Prepare the Plugin for Upload
1.	Archive the folder (mc-cs-plugin-custom-objects) into a ZIP file.
2.	Rename the ZIP file to CustomObjectsBundle.zip.
________________________________________
Step 3: Upload and Extract on Mautic Server
1.	Upload CustomObjectsBundle.zip to the plugins folder of your Mautic installation using SFTP or a file manager.
2.	Extract the ZIP file inside the plugins folder.
3.	Rename the extracted folder to CustomObjectsBundle.
________________________________________
Step 4: Install the Plugin in Mautic
1.	SSH into your Mautic server and navigate to the plugins directory
 or in file manager of your server just go to mautic installation directory and go to plugins folder.


`cd your-mautic-path/plugins`

IMPORTANT STEP: I had error 500, to fix it you need to manually run this command:

2.	Run the Mautic plugin installation command:

`sudo /usr/bin/php your-mautic-path/bin/console mautic:plugins:install`

o	You should see output confirming that 1 plugin was installed.

3.	Clear the Mautic cache:

`sudo /usr/bin/php your-mautic-path/bin/console cache:clear`
________________________________________
Step 5: Verify Plugin Installation
1.	Log in to your Mautic dashboard.
2.	Go to Settings > Plugins and check if the Custom Objects plugin is listed and activated.
________________________________________
Final Notes
•	If you encounter a 500 Internal Server Error after extracting the plugin, this is expected as Mautic needs to recognize the new plugin. It should be resolved after running the installation and clearing the cache.
•	If you experience any issues, check the Mautic logs in:

`your-mautic-path/var/logs/`

After all this you will have CustomObjectsBundle running for Mautic 5 version.

volha-pivavarchyk committed:
Get the export directory using the config

volha-pivavarchyk committed:
CS fixing

volha-pivavarchyk committed:
Remove unnesessary constant definition

volha-pivavarchyk committed:
Fix tests

volha-pivavarchyk committed:
Use master request instead of deprecated

volha-pivavarchyk committed:
Change phpunit configuration

volha-pivavarchyk committed:
CS fixing

volha-pivavarchyk commented:
@escopecz I fixed the problems after analyzing the cs fixer program. So far, three (cs, twig, phpunit) of the five tasks are working fine locally. Phpstan and rector failed. Do I need to fix all of them? Or is it possible to create a baseline for phpstan for example?

escopecz commented:
@volha-pivavarchyk yes, let's generate a baseline file for the PHPSTAN issues.

volha-pivavarchyk committed:
PHPstan fixes

volha-pivavarchyk committed:
PHPStan fixes some ORM deprecations

volha-pivavarchyk committed:
Add phpstan baseline

volha-pivavarchyk commented:
@escopecz Could you please approve CI running

volha-pivavarchyk committed:
Fix rector issues

volha-pivavarchyk committed:
Update phpstan baseline

volha-pivavarchyk committed:
Fix rector issues

volha-pivavarchyk committed:
Fix tests after fixing rector failures

volha-pivavarchyk committed:
Fix rector failures

volha-pivavarchyk committed:
Temporary set a parameter to allow the pipeline to continue even if Rector fails

volha-pivavarchyk commented:
@escopecz Two problems are left in CI.

1. One test fails. Locally, this test passes. The reason for the failure is that the `media/files/temp/custom_items_export_2024_12_11_23_52_22.csv` file could not be created.
```
$handler = @fopen($this->filePath, 'ab+');
fputcsv($handler, $header);
```
As a solution the following step can be created in the CI. After that test will pass in CI.
```
    - name: Create writable directory
      run: mkdir -p ./media/files/temp && chmod -R 755 ./media/files/temp
```
WDYT? Do you have another solution for it?

2. Two Rector checks are failing. I temporarily added `continue-on-error: true` to allow other steps to run and check their behavior. I will remove this setting after resolving the failures.

Rector suggests removing two parts of the code, but I'm not sure about it. I attempted to use comments to skip these failures, but it didn’t work. Is there a way to skip these failures, or must they be fixed? :blush:

volha-pivavarchyk committed:
Remove the CI Rector's temporary setting

volha-pivavarchyk committed:
Fix rector issues

volha-pivavarchyk committed:
Add a CI setting to crate a writable directory

volha-pivavarchyk commented:
@escopecz Could you please approve CI running

One question. For fixing the failed `testCustomItemExport` test I added a new step to CI. This test works well locally and failed on CI. The reason for the failure is that the `media/files/temp/custom_items_export_2024_12_11_23_52_22.csv` file could not be created.  So I just added the directory. 
```
- name: Create writable directory
      run: mkdir -p ./media/files/temp && chmod -R 755 ./media/files/temp
```
WDYT?

driskell commented:
@volha-pivavarchyk I opened https://github.com/mautic/mautic/pull/14437 to fix the tabs not appearing on Contacts when loaded via AJAX. The viewName was missing.

kuzmany reviewed (approved):
We are using this on our branch with couple of our modifications.
It works as expected. 
I recommend merge it to 5.x and continue improving it directly.

volha-pivavarchyk commented:
@driskell Thanks! I'll test it this evening
